src/utils/promptFilename.ts

```diff
@@ -0,0 +1,57 @@
+import * as vscode from "vscode";
+
+export const generateFriendlyPromptName = (promptType: string) => {
+    let friendlyPromptName = promptType;
+
+    if (friendlyPromptName.startsWith("github:")) {
+        const repo = friendlyPromptName.split("github:")[1].split("?")[0].replace("/", "-");
+        const path = friendlyPromptName.split("path=")[1].split("&")[0].replaceAll("/", "-");
+        friendlyPromptName = `${repo}-${path || 'root'}`;
+    }
+    else if (friendlyPromptName.startsWith("local:")) {
+        friendlyPromptName = friendlyPromptName.split("local://")[1].replaceAll("/", "-");
+    }
+    return friendlyPromptName;
+};
+
+export const preparePromptFile = async (workspaceFolder: vscode.WorkspaceFolder, promptType: string) => {
```

Comment:
I found it hard to understand what this function was at first.  Could we name this preparePromptOutputBuffer or something instead?